test/model/src/driver/kv_workload.cpp

```diff
@@ -37,6 +37,168 @@ extern "C" {
 
 namespace model {
 
+namespace operation {
+
+/*
+ * parse --
+ *     Parse an operation from a string. Throw an exception on error.
+ */
+any
+parse(const char *str)
+{
+    const char *p = str;
+
+    /* Get the operation name. */
+    std::string name;
+    while (*p != '\0' && (std::isalnum(*p) || *p == '_'))
+        name += *(p++);
+
+    while (*p != '\0' && std::isspace(*p))
+        p++;
+    if (*(p++) != '(')
+        throw model_exception("Expected '('");
+
+    /* Get the arguments. */
+    std::vector<std::string> args;
+    std::string current;
+    bool done = false;
+    bool escaped = false;
+    bool had_quotes = false;
+    bool quotes = false;
+    while (*p != '\0') {
+        char c = *(p++);
```

Comment:
Better: char c = *p++;

Reply:
I prefer to keep the parentheses there for readability, so that it is clear whether * or ++ is applied first.